django/db/models/expressions.py

```diff
@@ -702,7 +702,10 @@ def as_sql(self, compiler, connection, function=None, template=None, arg_joiner=
         sql_parts = []
         params = []
         for arg in self.source_expressions:
-            arg_sql, arg_params = compiler.compile(arg)
+            try:
+                arg_sql, arg_params = compiler.compile(arg)
+            except EmptyResultSet:
+                arg_sql, arg_params = getattr(arg, 'empty_aggregate_value', '0'), ()
```

Comment:
This won't work when `empty_aggregate_value` is `NotImplemented`.

Reply:
Also, we should re-raise an exception to the compiler when `empty_aggregate_value` is not implemented, e.g.
```python
except EmptyResultSet:
    arg_sql, arg_params = getattr(arg, 'empty_aggregate_value', NotImplemented), ()
    if arg_sql is NotImplemented:
        raise
```